Website/homepage 2025

nasin opened the pull request:
Hi,

Here's the demo viewer for the new homepage that we're looking to launch very soon.

Thanks!

TorsteinHonsi reviewed (approved):
Thanks, looks smashing!

nasin commented:
Hi, I made the following changes for the a11y stuff:

**For item 1:** 
- When the carousel is auto advancing, the charts are now hidden from screen readers.
- I changed the aria label for the carousel to "Highcharts product demos. Pause the carousel to explore demos individually and use the carousel controls to switch between demos." When the user pauses the carosel, the label changes to "Demo carousel paused. You can now explore individual demos using keyboard navigation."
- I added the hidden helper button to toggle the carousel between pause and playing

**For item 2:** When the user pauses the carousel, now all animations stop.

**For item 3:** I implemented the prefers-reduced-motion for the chart animations and the transitions between carousel slides

**For item 4:** I added the empty alt tag, and since all of the product titles are in a list that scrolls vertically with just the current one being visible, I added some logic to hide the "off screen" product  titles and just keep the current one visible to screen readers. I also hid the demo info in the footer since the chart description is more informative and that footer copy is more "decorative."

marvin19 commented:
For the funkiness, I think I've managed to recreate it if I hit resume while having reduce motion set in the OS. I guess you should be allowed to view the animation if you hit the resume button because then you are taking an active choice.

On the flight map, we could also do some point description to the series that describes the connection. Right now the aria-label says something like: `x, 9, . Series 2.`, When it could say 'Connection between Brazil and La Reunion' instead.

Also for the other point description, maybe include a piece of text saying 'Brazil, Arrived: 1770'.

The dots in the carousel are missing focus indicator.

Other than that, these are great improvements and we are now really close to being done 🙌🏻

nasin commented:
Hi,

Regarding the chart animations being weird...I went back to offering a link next to the title of each chart that lets the user  start or stop the chart animation. This way, users can control whether they want to see the chart animation independent of the pause/resume button on the carousel.

So the pause/resume button controls the carousel, and the individual charts with animations get their own stop/start chart animation links. This means when you click the pause/resume button, only the carousel is affected, and charts will obey the user preference.

I added announcment functionality for when chart animations are started or stopped.

For reduced motion users, the carousel and the chart animations are all paused on load. The reduced motion user can then opt to animate individual charts by clicking the "start chart animation" link next to each demo title (where appropriate.)

Also, the carousel now keeps track of which chart animations have been stopped or started by the user, so when the chart comes back around, the chart will be animated or stopped depending on user preference independent of the carousel state. This behavior applies to all users. 

I added the point descriptions for the mapline and mappoint series.

The dots should now get focus when tabbed to.
